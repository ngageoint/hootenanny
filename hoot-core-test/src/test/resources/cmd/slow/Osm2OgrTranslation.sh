#!/bin/bash

OUTPUT=test-output/cmd/slow/Osm2OgrTranslation

function getLayerNames() {
    result=`ogrinfo -q $1 2>&1 | sed "s/[0-9]*: //;s/ \(.*\)//g" | sort`
}

# Using this function instead of "ogrinfo -al" gives more consistent output between
# version 1.9 and 1.10 of GDAL. If this is failing it may be due to a formatting
# change in GDAL
function printLayerInfo() {
    path=$1
    getLayerNames $path
    layerNames=$result
    for l in $layerNames
    do
        echo $l
        ogrinfo -q -geom=SUMMARY $path $l | sed "s/  */ /g"
    done
}

rm -rf $OUTPUT
mkdir -p $OUTPUT

echo "#### Shapefile Output ####"
hoot --osm2ogr test-files/io/SampleTranslation.js test-files/io/SampleTranslation.osm $OUTPUT/output.shp
printLayerInfo $OUTPUT/output/

echo "#### FileGDB Output ####"
hoot --osm2ogr test-files/io/SampleTranslation.js test-files/io/SampleTranslation.osm $OUTPUT/output.gdb
printLayerInfo $OUTPUT/output.gdb/

echo "#### Test prepending layer names and lazy layer creation ####"
hoot --osm2ogr -D ogr.writer.pre.layer.name=bar_ test-files/io/SampleTranslation.js test-files/io/SampleTranslation.osm $OUTPUT/options.shp
hoot --osm2ogr -D ogr.writer.pre.layer.name=foo_ -D ogr.writer.create.all.layers=true test-files/io/SampleTranslation.js test-files/io/SampleTranslation.osm $OUTPUT/options.shp
printLayerInfo $OUTPUT/options/

echo "#### Test to make sure all layers are read from a data source. ####"
rm -rf test-output/cmd/slow/delaware-tds test-output/cmd/slow/delaware-tds.shp
mkdir -p test-output/cmd/slow
hoot osm2ogr test-files/io/O2sTranslation.js test-files/cmd/slow/delaware.shp test-output/cmd/slow/delaware-tds.shp
mv test-output/cmd/slow/delaware-tds test-output/cmd/slow/delaware-tds.shp
hoot stats --quick test-output/cmd/slow/delaware-tds.shp

