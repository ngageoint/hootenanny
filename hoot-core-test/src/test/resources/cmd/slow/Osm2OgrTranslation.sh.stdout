#### Shapefile Output ####
AAL015

Layer name: AAL015
OGRFeature(AAL015):0
 NAM (String) = w1
 ARA (Real) = 10.000000000000000
 POLYGON : 4 points

OGRFeature(AAL015):1
 NAM (String) = r1
 ARA (Real) = 10.000000000000000
 POLYGON : 4 points, 1 inner rings (4 points)

LAP010

Layer name: LAP010
OGRFeature(LAP010):0
 NAM (String) = w2
 ARA (Real) = -999999.000000000000000
 PCF (Integer) = 1
 LINESTRING : 2 points

LAP030

Layer name: LAP030
OGRFeature(LAP030):0
 NAM (String) = w3
 ARA (Real) = -999999.000000000000000
 PCF (Integer) = 2
 LTN (Integer) = 2
 LINESTRING : 2 points

PAL015

Layer name: PAL015
OGRFeature(PAL015):0
 NAM (String) = n1
 ARA (Real) = -999999.000000000000000
 POINT : 

#### FileGDB Output ####
AAL015

Layer name: AAL015
OGRFeature(AAL015):1
 NAM (String) = w1
 ARA (Real) = 10
 MULTIPOLYGON : 1 geometries:
 POLYGON : 4 points

OGRFeature(AAL015):2
 NAM (String) = r1
 ARA (Real) = 10
 MULTIPOLYGON : 1 geometries:
 POLYGON : 4 points, 1 inner rings (4 points)

LAP010

Layer name: LAP010
OGRFeature(LAP010):1
 NAM (String) = w2
 ARA (Real) = -999999
 PCF (Integer) = 1
 MULTILINESTRING : 1 geometries:
 LINESTRING : 2 points

LAP030

Layer name: LAP030
OGRFeature(LAP030):1
 NAM (String) = w3
 ARA (Real) = -999999
 PCF (Integer) = 2
 LTN (Integer) = 2
 MULTILINESTRING : 1 geometries:
 LINESTRING : 2 points

PAL015

Layer name: PAL015
OGRFeature(PAL015):1
 NAM (String) = n1
 ARA (Real) = -999999
 POINT : 

#### Test prepending layer names and lazy layer creation ####
bar_AAL015

Layer name: bar_AAL015
OGRFeature(bar_AAL015):0
 NAM (String) = w1
 ARA (Real) = 10.000000000000000
 POLYGON : 4 points

OGRFeature(bar_AAL015):1
 NAM (String) = r1
 ARA (Real) = 10.000000000000000
 POLYGON : 4 points, 1 inner rings (4 points)

bar_LAP010

Layer name: bar_LAP010
OGRFeature(bar_LAP010):0
 NAM (String) = w2
 ARA (Real) = -999999.000000000000000
 PCF (Integer) = 1
 LINESTRING : 2 points

bar_LAP030

Layer name: bar_LAP030
OGRFeature(bar_LAP030):0
 NAM (String) = w3
 ARA (Real) = -999999.000000000000000
 PCF (Integer) = 2
 LTN (Integer) = 2
 LINESTRING : 2 points

bar_PAL015

Layer name: bar_PAL015
OGRFeature(bar_PAL015):0
 NAM (String) = n1
 ARA (Real) = -999999.000000000000000
 POINT : 

foo_AAL015

Layer name: foo_AAL015
OGRFeature(foo_AAL015):0
 NAM (String) = w1
 ARA (Real) = 10.000000000000000
 POLYGON : 4 points

OGRFeature(foo_AAL015):1
 NAM (String) = r1
 ARA (Real) = 10.000000000000000
 POLYGON : 4 points, 1 inner rings (4 points)

foo_LAP010

Layer name: foo_LAP010
OGRFeature(foo_LAP010):0
 NAM (String) = w2
 ARA (Real) = -999999.000000000000000
 PCF (Integer) = 1
 LINESTRING : 2 points

foo_LAP020

Layer name: foo_LAP020
foo_LAP030

Layer name: foo_LAP030
OGRFeature(foo_LAP030):0
 NAM (String) = w3
 ARA (Real) = -999999.000000000000000
 PCF (Integer) = 2
 LTN (Integer) = 2
 LINESTRING : 2 points

foo_PAL015

Layer name: foo_PAL015
OGRFeature(foo_PAL015):0
 NAM (String) = n1
 ARA (Real) = -999999.000000000000000
 POINT : 

#### Test to make sure all layers are read from a data source. ####
10:53:48.087 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"183418011","error:circular":"15","hoot:layername":"planet_osm_point","hoot:status":"invalid","z_order":"0","railway":"level_crossing"}
10:53:48.088 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"183410950","error:circular":"15","hoot:layername":"planet_osm_point","hoot:status":"invalid","z_order":"0","railway":"level_crossing"}
10:53:48.088 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"183292835","error:circular":"15","hoot:layername":"planet_osm_point","hoot:status":"invalid","z_order":"0","railway":"level_crossing"}
10:53:48.088 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"183430685","error:circular":"15","hoot:layername":"planet_osm_point","hoot:status":"invalid","z_order":"0","railway":"level_crossing"}
10:53:48.089 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"17709194","error:circular":"15","hoot:layername":"planet_osm_roads","oneway":"yes","hoot:status":"invalid","z_order":"8","tags":"\"NHS\"=>\"STRAHNET\", \"hgv\"=>\"designated\", \"lanes\"=>\"2\", \"name_1\"=>\"Ocean Highway\", \"tiger:cfcc\"=>\"A25\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:reviewed\"=>\"no\", \"tiger:zip_left\"=>\"21875\", \"tiger:name_base\"=>\"Salisbury\", \"tiger:name_type\"=>\"Blvd\", \"tiger:nam","name":"North Salisbury Boulevard","ref":"US 13","way_area":"0","highway":"trunk"}
10:53:48.089 WARN  ...?(  -1) CPL Error #: 1 Value '{"osm_id":"17709194","error:circular":"15","hoot:layername":"planet_osm_roads","oneway":"yes","hoot:status":"invalid","z_order":"8","tags":"\"NHS\"=>\"STRAHNET\", \"hgv\"=>\"designated\", \"lanes\"=>\"2\", \"name_1\"=>\"Ocean Highway\", \"tiger:cfcc\"=>\"A25\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:reviewed\"=>\"no\", \"tiger:zip_left\"=>\"21875\", \"tiger:name_base\"=>\"Salisbury\", \"tiger:name_type\"=>\"Blvd\", \"tiger:nam","name":"North Salisbury Boulevard","ref":"US 13","way_area":"0","highway":"trunk"}' of field tags has been truncated to 254 characters.
This warning will not be emitted any more for that layer.
10:53:48.090 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"17709181","error:circular":"15","hoot:layername":"planet_osm_roads","oneway":"yes","hoot:status":"invalid","z_order":"8","tags":"\"NHS\"=>\"STRAHNET\", \"hgv\"=>\"designated\", \"lanes\"=>\"2\", \"name_1\"=>\"Ocean Highway\", \"tiger:cfcc\"=>\"A25\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:reviewed\"=>\"no\", \"tiger:name_base\"=>\"Salisbury\", \"tiger:name_type\"=>\"Blvd\", \"tiger:name_base_1\"=>\"Ocean\", \"tiger:","name":"North Salisbury Boulevard","ref":"US 13","way_area":"0","highway":"trunk"}
10:53:48.090 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"34320466","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_roads","boundary":"administrative","hoot:status":"invalid","z_order":"0","way_area":"0"}
10:53:48.092 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"place":"town","osm_id":"-2057485","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_roads","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"is_in\"=>\"USA, Maryland\", \"tiger:CPI\"=>\"N\", \"wikipedia\"=>\"en:Delmar, Maryland\", \"tiger:LSAD\"=>\"43\", \"tiger:NAME\"=>\"Delmar\", \"border_type\"=>\"town\", \"is_in:state\"=>\"Maryland\", \"tiger:MTFCC\"=>\"G4110\", \"is_in:country\"=>\"USA\", \"tiger:CLASSFP\"=>\"C1\", \"tiger:PC","name":"Delmar","way_area":"0"}
10:53:48.094 WARN  ...?(  -1) CPL Error #: 1 Value '{"place":"town","osm_id":"-2057485","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_roads","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"is_in\"=>\"USA, Maryland\", \"tiger:CPI\"=>\"N\", \"wikipedia\"=>\"en:Delmar, Maryland\", \"tiger:LSAD\"=>\"43\", \"tiger:NAME\"=>\"Delmar\", \"border_type\"=>\"town\", \"is_in:state\"=>\"Maryland\", \"tiger:MTFCC\"=>\"G4110\", \"is_in:country\"=>\"USA\", \"tiger:CLASSFP\"=>\"C1\", \"tiger:PC","name":"Delmar","way_area":"0"}' of field tags has been truncated to 254 characters.
This warning will not be emitted any more for that layer.
10:53:48.094 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"34320474","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_roads","boundary":"administrative","hoot:status":"invalid","z_order":"0","way_area":"0"}
10:53:48.095 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"place":"town","osm_id":"-2057485","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_roads","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"is_in\"=>\"USA, Maryland\", \"tiger:CPI\"=>\"N\", \"wikipedia\"=>\"en:Delmar, Maryland\", \"tiger:LSAD\"=>\"43\", \"tiger:NAME\"=>\"Delmar\", \"border_type\"=>\"town\", \"is_in:state\"=>\"Maryland\", \"tiger:MTFCC\"=>\"G4110\", \"is_in:country\"=>\"USA\", \"tiger:CLASSFP\"=>\"C1\", \"tiger:PC","name":"Delmar","way_area":"0"}
10:53:48.096 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"169849446","error:circular":"15","hoot:layername":"planet_osm_roads","operator":"NS","hoot:status":"invalid","z_order":"5","tags":"\"gauge\"=>\"1435\", \"tiger:cfcc\"=>\"B11\", \"tiger:tlid\"=>\"106074354:106074356:106074358:106074363:10609:106092347:106092349:106092542:106092668:106100243:106100973:106104568:106104569:106107231:106107232:106107872:106107873:106108164:106108165:106110525:10611","name":"Delmarva Secondary","railway":"rail","way_area":"0"}
10:53:48.097 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"-934717","error:circular":"15","admin_leve":"6","hoot:layername":"planet_osm_roads","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"wikipedia\"=>\"en:Wicomico County, Maryland\", \"attribution\"=>\"USGS 2001 County Boundary\", \"border_type\"=>\"county\", \"nist:fips_code\"=>\"24045\", \"nist:state_fips\"=>\"24\"","name":"Wicomico County","way_area":"0"}
10:53:48.098 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"-162112","error:circular":"15","admin_leve":"4","hoot:layername":"planet_osm_roads","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"name:ar\"=>\"ÙØ§Ø±ÙÙØ§ÙØ¯\", \"name:az\"=>\"Merilend\", \"name:bg\"=>\"ÐÐµÑÐ¸Ð»ÐµÐ½Ð´\", \"name:bn\"=>\"à¦®à§à¦°à¦¿à¦²à§à¦¯à¦¾à¦¨à§à¦¡\", \"name:el\"=>\"ÎÎ­ÏÎ¹Î»Î±Î½Ï\", \"name:en\"=>\"Maryland\", \"name:eo\"=>\"Marilando\", \"name:fa\"=>\"ÙØ±ÛÙÙØ¯\", \"name:fy\"=>\"Mary","name":"Maryland","ref":"MD","way_area":"0"}
10:53:48.100 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"place":"town","osm_id":"-2057485","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_polygon","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"is_in\"=>\"USA, Maryland\", \"tiger:CPI\"=>\"N\", \"wikipedia\"=>\"en:Delmar, Maryland\", \"tiger:LSAD\"=>\"43\", \"tiger:NAME\"=>\"Delmar\", \"border_type\"=>\"town\", \"is_in:state\"=>\"Maryland\", \"tiger:MTFCC\"=>\"G4110\", \"is_in:country\"=>\"USA\", \"tiger:CLASSFP\"=>\"C1\", \"tiger:PC","name":"Delmar","area":"yes","way_area":"0.000252"}
10:53:48.102 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"311461170","error:circular":"15","hoot:layername":"planet_osm_polygon","hoot:status":"invalid","z_order":"0","area":"yes","landuse":"residential","way_area":"0.00016799999999999999"}
10:53:48.102 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"place":"town","osm_id":"-2057485","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_polygon","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"is_in\"=>\"USA, Maryland\", \"tiger:CPI\"=>\"N\", \"wikipedia\"=>\"en:Delmar, Maryland\", \"tiger:LSAD\"=>\"43\", \"tiger:NAME\"=>\"Delmar\", \"border_type\"=>\"town\", \"is_in:state\"=>\"Maryland\", \"tiger:MTFCC\"=>\"G4110\", \"is_in:country\"=>\"USA\", \"tiger:CLASSFP\"=>\"C1\", \"tiger:PC","name":"Delmar","area":"yes","way_area":"0.000192"}
10:53:48.104 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"17704553","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"3","tags":"\"tiger:cfcc\"=>\"A41\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:reviewed\"=>\"no\", \"tiger:name_base\"=>\"Stage\", \"tiger:name_type\"=>\"Rd\"","name":"Stage Road","way_area":"0","highway":"residential"}
10:53:48.104 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"17709194","error:circular":"15","hoot:layername":"planet_osm_line","oneway":"yes","hoot:status":"invalid","z_order":"8","tags":"\"NHS\"=>\"STRAHNET\", \"hgv\"=>\"designated\", \"lanes\"=>\"2\", \"name_1\"=>\"Ocean Highway\", \"tiger:cfcc\"=>\"A25\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:reviewed\"=>\"no\", \"tiger:zip_left\"=>\"21875\", \"tiger:name_base\"=>\"Salisbury\", \"tiger:name_type\"=>\"Blvd\", \"tiger:nam","name":"North Salisbury Boulevard","ref":"US 13","way_area":"0","highway":"trunk"}
10:53:48.105 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"-2301100","route":"road","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"0","tags":"\"symbol\"=>\"http://upload.wikimedia.org/wikipedia/commons/e/e8/US_13.svg\", \"network\"=>\"US:US\", \"wikipedia\"=>\"en:U.S. Route 13 in Maryland\", \"route_name\"=>\"US 13 (MD)\", \"is_in:state\"=>\"MD\", \"route_pref_color\"=>\"0\"","ref":"13","way_area":"0"}
10:53:48.106 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"17709181","error:circular":"15","hoot:layername":"planet_osm_line","oneway":"yes","hoot:status":"invalid","z_order":"8","tags":"\"NHS\"=>\"STRAHNET\", \"hgv\"=>\"designated\", \"lanes\"=>\"2\", \"name_1\"=>\"Ocean Highway\", \"tiger:cfcc\"=>\"A25\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:reviewed\"=>\"no\", \"tiger:name_base\"=>\"Salisbury\", \"tiger:name_type\"=>\"Blvd\", \"tiger:name_base_1\"=>\"Ocean\", \"tiger:","name":"North Salisbury Boulevard","ref":"US 13","way_area":"0","highway":"trunk"}
10:53:48.106 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"-2301100","route":"road","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"0","tags":"\"symbol\"=>\"http://upload.wikimedia.org/wikipedia/commons/e/e8/US_13.svg\", \"network\"=>\"US:US\", \"wikipedia\"=>\"en:U.S. Route 13 in Maryland\", \"route_name\"=>\"US 13 (MD)\", \"is_in:state\"=>\"MD\", \"route_pref_color\"=>\"0\"","ref":"13","way_area":"0"}
10:53:48.107 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"85462125","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"3","tags":"\"tiger:cfcc\"=>\"A41\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:reviewed\"=>\"no\", \"tiger:name_base\"=>\"2nd\", \"tiger:name_type\"=>\"St\", \"tiger:name_direction_prefix\"=>\"S\"","name":"South 2nd Street","way_area":"0","highway":"residential"}
10:53:48.108 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"17705776","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"3","tags":"\"tiger:cfcc\"=>\"A41\", \"tiger:tlid\"=>\"114446919:114446895:114446939\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:source\"=>\"tiger_import_dch_v0.6_20070813\", \"tiger:reviewed\"=>\"no\", \"tiger:name_base\"=>\"1st\", \"tiger:name_type\"=>\"St\", \"tiger:separated\"=>\"no\", \"tig","name":"South 1st Street","way_area":"0","highway":"residential"}
10:53:48.108 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"17707651","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"3","tags":"\"tiger:cfcc\"=>\"A41\", \"tiger:tlid\"=>\"114446918:114446893:114446958:114446938:114446866:114448029:114484855:114484856\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:source\"=>\"tiger_import_dch_v0.6_20070813\", \"tiger:reviewed\"=>\"no\", \"tiger:name_base\"=>\"Pennsylvan","name":"South Pennsylvania Avenue","way_area":"0","highway":"residential"}
10:53:48.108 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"place":"town","osm_id":"-2057485","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_line","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"is_in\"=>\"USA, Maryland\", \"tiger:CPI\"=>\"N\", \"wikipedia\"=>\"en:Delmar, Maryland\", \"tiger:LSAD\"=>\"43\", \"tiger:NAME\"=>\"Delmar\", \"border_type\"=>\"town\", \"is_in:state\"=>\"Maryland\", \"tiger:MTFCC\"=>\"G4110\", \"is_in:country\"=>\"USA\", \"tiger:CLASSFP\"=>\"C1\", \"tiger:PC","name":"Delmar","way_area":"0"}
10:53:48.110 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"34320466","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_line","boundary":"administrative","hoot:status":"invalid","z_order":"0","way_area":"0"}
10:53:48.113 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"34320474","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_line","boundary":"administrative","hoot:status":"invalid","z_order":"0","way_area":"0"}
10:53:48.114 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"place":"town","osm_id":"-2057485","error:circular":"15","admin_leve":"8","hoot:layername":"planet_osm_line","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"is_in\"=>\"USA, Maryland\", \"tiger:CPI\"=>\"N\", \"wikipedia\"=>\"en:Delmar, Maryland\", \"tiger:LSAD\"=>\"43\", \"tiger:NAME\"=>\"Delmar\", \"border_type\"=>\"town\", \"is_in:state\"=>\"Maryland\", \"tiger:MTFCC\"=>\"G4110\", \"is_in:country\"=>\"USA\", \"tiger:CLASSFP\"=>\"C1\", \"tiger:PC","name":"Delmar","way_area":"0"}
10:53:48.115 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"169849446","error:circular":"15","hoot:layername":"planet_osm_line","operator":"NS","hoot:status":"invalid","z_order":"5","tags":"\"gauge\"=>\"1435\", \"tiger:cfcc\"=>\"B11\", \"tiger:tlid\"=>\"106074354:106074356:106074358:106074363:10609:106092347:106092349:106092542:106092668:106100243:106100973:106104568:106104569:106107231:106107232:106107872:106107873:106108164:106108165:106110525:10611","name":"Delmarva Secondary","railway":"rail","way_area":"0"}
10:53:48.116 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"-2319895","route":"road","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"0","tags":"\"NHS\"=>\"STRAHNET\", \"route_pref_color\"=>\"0\"","way_area":"0"}
10:53:48.118 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"109829734","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"4","tags":"\"tiger:cfcc\"=>\"A41\", \"tiger:tlid\"=>\"114445550:114460681:114485462:114485464:114485466:114485467\", \"tiger:county\"=>\"Wicomico, MD\", \"tiger:source\"=>\"tiger_import_dch_v0.6_20070813\", \"tiger:reviewed\"=>\"no\", \"tiger:name_base\"=>\"Jersey\", \"tiger:name_type\"=>\"R","name":"Jersey Road","way_area":"0","highway":"tertiary"}
10:53:48.119 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"17707365","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"4","tags":"\"odbl\"=>\"clean\", \"odbl:note\"=>\"All changes made by non-CT-accepting contributors have been reverted.\", \"tiger:cfcc\"=>\"A41\", \"tiger:tlid\"=>\"114478990:114478991:114478998:114494961:114479002:114480720:114480721:114480671:114480672:114480723:114480724:10613","name":"Waller Road","way_area":"0","highway":"tertiary"}
10:53:48.120 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"-2319895","route":"road","error:circular":"15","hoot:layername":"planet_osm_line","hoot:status":"invalid","z_order":"0","tags":"\"NHS\"=>\"STRAHNET\", \"route_pref_color\"=>\"0\"","way_area":"0"}
10:53:48.128 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"-934717","error:circular":"15","admin_leve":"6","hoot:layername":"planet_osm_line","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"wikipedia\"=>\"en:Wicomico County, Maryland\", \"attribution\"=>\"USGS 2001 County Boundary\", \"border_type\"=>\"county\", \"nist:fips_code\"=>\"24045\", \"nist:state_fips\"=>\"24\"","name":"Wicomico County","way_area":"0"}
10:53:48.129 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"-2258990","route":"railway","error:circular":"15","hoot:layername":"planet_osm_line","operator":"NS","hoot:status":"invalid","z_order":"0","tags":"\"route_name\"=>\"Delmarva Secondary\", \"route_pref_color\"=>\"0\"","way_area":"0"}
10:53:48.133 INFO  ...e-1/hoot/rules/lib/HootLib.js"( 252) {"osm_id":"-162112","error:circular":"15","admin_leve":"4","hoot:layername":"planet_osm_line","boundary":"administrative","hoot:status":"invalid","z_order":"0","tags":"\"name:ar\"=>\"ÙØ§Ø±ÙÙØ§ÙØ¯\", \"name:az\"=>\"Merilend\", \"name:bg\"=>\"ÐÐµÑÐ¸Ð»ÐµÐ½Ð´\", \"name:bn\"=>\"à¦®à§à¦°à¦¿à¦²à§à¦¯à¦¾à¦¨à§à¦¡\", \"name:el\"=>\"ÎÎ­ÏÎ¹Î»Î±Î½Ï\", \"name:en\"=>\"Maryland\", \"name:eo\"=>\"Marilando\", \"name:fa\"=>\"ÙØ±ÛÙÙØ¯\", \"name:fy\"=>\"Mary","name":"Maryland","ref":"MD","way_area":"0"}
Reprojecting 0 / 5664         Reprojecting 1000 / 5664         Reprojecting 2000 / 5664         Reprojecting 3000 / 5664         Reprojecting 4000 / 5664         Reprojecting 5000 / 5664         
10:53:48.318 INFO  .../core/ops/CalculateStatsOp.cpp(  92) Calculating map statistics...
10:53:48.320 INFO  .../core/ops/CalculateStatsOp.cpp( 523) Map statistics calculated.
Stat Name	test-output/cmd/slow/delaware-tds.shp
Node Count	5664
Way Count	32
Relation Count	0
Minimum Node ID	-5664
Maximum Node ID	-1
Minimum Way ID	-32
Maximum Way ID	-1
Minimum Relation ID	9.2233720368547758e+18
Maximum Relation ID	-9.2233720368547758e+18
